pkg/network/tracer/tracer.go

```diff
@@ -303,55 +286,30 @@ func newReverseDNS(c *config.Config) dns.ReverseDNS {
 	return rdns
 }
 
-func runOffsetGuessing(config *config.Config, buf bytecode.AssetReader) ([]manager.ConstantEditor, error) {
-	// Enable kernel probes used for offset guessing.
-	offsetMgr := newOffsetManager()
-	offsetOptions := manager.Options{
-		RLimit: &unix.Rlimit{
-			Cur: math.MaxUint64,
-			Max: math.MaxUint64,
-		},
-	}
-	enabledProbes, err := offsetGuessProbes(config)
-	if err != nil {
-		return nil, fmt.Errorf("unable to configure offset guessing probes: %w", err)
-	}
-
-	for _, p := range offsetMgr.Probes {
-		if _, enabled := enabledProbes[p.EBPFFuncName]; !enabled {
-			offsetOptions.ExcludedFunctions = append(offsetOptions.ExcludedFunctions, p.EBPFFuncName)
-		}
-	}
-	for funcName := range enabledProbes {
-		offsetOptions.ActivatedProbes = append(
-			offsetOptions.ActivatedProbes,
-			&manager.ProbeSelector{
-				ProbeIdentificationPair: manager.ProbeIdentificationPair{
-					EBPFFuncName: funcName,
-					UID:          "offset",
-				},
-			})
-	}
-	if err := offsetMgr.InitWithOptions(buf, offsetOptions); err != nil {
-		return nil, fmt.Errorf("could not load bpf module for offset guessing: %s", err)
+func runOffsetGuessing(config *config.Config, buf bytecode.AssetReader, guesser offsetguess.OffsetGuesser) (editors []manager.ConstantEditor, err error) {
+	if err := offsetguess.SetupOffsetGuesser(guesser, config, buf); err != nil {
+		return nil, err
 	}
 
-	if err := offsetMgr.Start(); err != nil {
-		return nil, fmt.Errorf("could not start offset ebpf manager: %s", err)
-	}
 	defer func() {
-		err := offsetMgr.Stop(manager.CleanAll)
+		err := guesser.Manager().Stop(manager.CleanAll)
 		if err != nil {
 			log.Warnf("error stopping offset ebpf manager: %s", err)
 		}
 	}()
-	start := time.Now()
-	editors, err := guessOffsets(offsetMgr, config)
-	if err != nil {
-		return nil, err
+
+	// Offset guessing has been flaky for some customers, so if it fails we'll retry it up to 5 times
+	for i := 0; i < 5; i++ {
+		start := time.Now()
+		if editors, err = guesser.Guess(config); err == nil {
```

Comment:
Will this use stale state from the previous run? Before, we would re-create everything.

Reply:
`Guess` does reset the offset guess state